test/integration/authorizeUsersAndService.test.ts

```diff
@@ -0,0 +1,182 @@
+    });
+
+    it("should call verifyDiscordBot for valid Discord bot token", async function () {
+      const jwtToken = bot.generateToken({ name: CLOUDFLARE_WORKER });
+      req.headers.authorization = `Bearer ${jwtToken}`;
+      await authorization([ROLES.APPOWNER], [Services.CLOUDFLARE_WORKER])(req, res, next);
+      expect(next.calledOnce).to.be.equal(true);
+    });
+
+    it("should return unauthorized for unknown service names", async function () {
+      const jwtToken = bot.generateToken({ name: "Invalid name" });
+      req.headers.authorization = `Bearer ${jwtToken}`;
+      await authorization([ROLES.APPOWNER], [Services.CLOUDFLARE_WORKER])(req, res, next);
+      expect(res.boom.unauthorized.calledOnce).to.be.equal(true);
+    });
+  });
+  describe("User Authorization", function () {
+    const pongHandler = (_: CustomRequest, res: CustomResponse) => {
+      return res.json({ message: "pong" });
+    };
+
+    router.get("/for-super-user", authorization([ROLES.SUPERUSER], [Services.CRON_JOB_HANDLER]), pongHandler);
+
+    const app = express();
+    AppMiddlewares(app);
+    app.use("/", router);
+
+    beforeEach(async () => {
+      const defaultUserId = await addUser(defaultUser);
+      const superUserId = await addUser(superUser);
+      defaultJwt = authService.generateAuthToken({ userId: defaultUserId });
+      superUserJwt = authService.generateAuthToken({ userId: superUserId });
+    });
+    afterEach(async () => {
+      await cleanDb();
+    });
+    it("should authorize super user for route with super_user required role", function (done) {
+      chai
+        .request(app)
+        .get("/for-super-user")
+        .set("cookie", `${cookieName}=${superUserJwt}`)
+        .end((err: Error, res: CustomResponse) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(200);
+          return done();
+        });
+    });
+    it("should not authenticate invalid tokens", function (done) {
+      const jwtToken = bot.generateCronJobToken({ name: CRON_JOB_HANDLER });
+      chai
+        .request(app)
+        .get("/for-super-user")
+        .set("cookie", `${cookieName}=${jwtToken}`)
+        .end((err: Error, res: CustomResponse) => {
+          if (err) {
+            return done(err);
+          }
+          expect(res).to.have.status(401);
+          return done();
+        });
+    });
+    it("should not allow default user for route with super_user required role", function (done) {
+      chai
+        .request(app)
+        .get("/for-super-user")
+        .set("cookie", `${cookieName}=${defaultJwt}`)
+        .end((err: Error, res: CustomResponse) => {
+          if (err) {
+            return done(err);
+          }
+          console.log(res);
+          expect(res).to.have.status(401);
+          return done();
+        });
+    });
+    it("should allow services if user does not have authorization", function (done) {
```

Comment:
Didn't get this test, would on-super user have access to services in `/for-super-user` ?

Reply:
Since we are also passing cron job token, it will be prioritised and since we have allowed that service for this route, auth should pass. 
In practice we won't be passing two different tokens together.